# Build Tiled Library

This task includes the following main steps:

* Tile library
* Calculate FSP and segment downstream distance
* Assign stream order to segments

## Import python packages

In [18]:
import sys
import time

# Set tools/scripts folder (YOU NEED TO CHANGE THIS)
fldplnToolFolder = r'E:/CUAHSI_SI/training/source' #r'C:\Users\lixi\OneDrive - The University of Kansas\FLDPLN\tools_os'

# add the tools folder to sys.path to access the fldpln module
sys.path.append(fldplnToolFolder) 
# fldpln modules
from fldpln_library import *
from fldpln import *

## Tile Library

With segment-based library as input, we spatially divide it into tiles based on the tile size provided. Note that tile size is the number of cells to avoid partial cells within a tile and can be used for both PCS and GCS. 

This process also copies the FSP and segment info CSV files to the tiled library and creates a metadata file (TileCellSizeSpatialReference.json) which stores library tile and cell sizes and spatial reference in a JSON file.

**Note that this is the most time consuming process and may take hours for large libraries.**

In [19]:
# Segment-based library
segLibFolder = r'E:\CUAHSI_SI\training\examples\wildcat_10m_3dep'

# tiled library folder
tiledLibFolder = r'E:\CUAHSI_SI\training\examples\wildcat_10m_3dep\tiled_snz_library' 

# define tile size (number of cells) and format
cellSize = 10
tileSize = 200 # number of cells
tileFileFormat = 'snappy' # 'snappy' or 'mat'

# libraries to be tiled. 
# Note that the tiled libraries will have the same name as the segment-based libraries except they are located under the segFolder!
libNames = ['lib_py'] # libs for Wildcat Creek

# tile libraries
for libName in libNames: 
    print(f'Tile library: {libName} ...')
    TileLibrary(os.path.join(segLibFolder,libName), cellSize, os.path.join(tiledLibFolder,libName),tileSize,tileFileFormat) 

Tile library: lib_py ...
Calculate library extent ...


E:\CUAHSI_SI/training/source\fldpln_library.py:313: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  segExts = pd.concat([segExts,segExt])


Library external border extent (minX, maxX, minY, maxY) : (-65353.2057939356, -39543.2057939356, 1783762.9445651, 1802122.9445651)
Total number of FSP-FPP relations: 2263154
Number of (possible) tiles: 130
Tile extents:
 [(-65353.2057939356, -63353.2057939356, 1783762.9445651, 1785762.9445651), (-65353.2057939356, -63353.2057939356, 1785762.9445651, 1787762.9445651), (-65353.2057939356, -63353.2057939356, 1787762.9445651, 1789762.9445651), (-65353.2057939356, -63353.2057939356, 1789762.9445651, 1791762.9445651), (-65353.2057939356, -63353.2057939356, 1791762.9445651, 1793762.9445651), (-65353.2057939356, -63353.2057939356, 1793762.9445651, 1795762.9445651), (-65353.2057939356, -63353.2057939356, 1795762.9445651, 1797762.9445651), (-65353.2057939356, -63353.2057939356, 1797762.9445651, 1799762.9445651), (-65353.2057939356, -63353.2057939356, 1799762.9445651, 1801762.9445651), (-65353.2057939356, -63353.2057939356, 1801762.9445651, 1803762.9445651), (-63353.2057939356, -61353.2057939356,

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:208: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determini

Total number of FSP-FPP relations in the tile: 0
Processing tile:  2
Tile extent (minX, maxX, minY, maxY) : (-65353.2057939356, -63353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  2
Tile extent (minX, maxX, minY, maxY) : (-65353.2057939356, -63353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  2
Tile extent (minX, maxX, minY, maxY) : (-65353.2057939356, -63353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  2
Tile extent (minX, maxX, minY, maxY) : (-65353.2057939356, -63353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  5
Total number of FSP-FPP relations in the tile: 0
Processing tile:  2
Tile extent

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Total number of FSP-FPP relations in the tile: 0
Processing tile:  4
Tile extent (minX, maxX, minY, maxY) : (-63353.2057939356, -61353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  4
Tile extent (minX, maxX, minY, maxY) : (-63353.2057939356, -61353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  4
Tile extent (minX, maxX, minY, maxY) : (-63353.2057939356, -61353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  4
Tile extent (minX, maxX, minY, maxY) : (-63353.2057939356, -61353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  4
Tile extent

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Total number of FSP-FPP relations in the tile: 2520
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-61353.2057939356, -60003.2057939356, 1784682.9445651, 1785762.9445651)
Processing tile:  8
Tile extent (minX, maxX, minY, maxY) : (-61353.2057939356, -59353.2057939356, 1785762.9445651, 1787762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 3059
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-60233.2057939356, -59353.2057939356, 1785762.9445651, 1787262.9445651)
Processing tile:  9
Tile extent (minX, maxX, minY, maxY) : (-6135

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Total number of FSP-FPP relations in the tile: 0
Processing tile:  9
Tile extent (minX, maxX, minY, maxY) : (-61353.2057939356, -59353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  9
Tile extent (minX, maxX, minY, maxY) : (-61353.2057939356, -59353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  9
Tile extent (minX, maxX, minY, maxY) : (-61353.2057939356, -59353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  10
Total number of FSP-FPP relations in the tile: 0
Processing tile:  9
Tile extent (minX, maxX, minY, maxY) : (-61353.2057939356, -59353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  9
Total number of FSP-FPP relations in the tile: 71423
Saving FSP-FPP relations i

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Total number of FSP-FPP relations in the tile: 35976
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 94
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-61663.2057939356, -57313.2057939356, 1796972.9445651, 1800132.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-61353.2057939356, -59853.2057939356, 1799762.9445651, 1800872.9445651)
Processing tile:  11
Tile extent (minX, maxX, minY, maxY) : (-61353.2057939356, -59353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interseting with the tile:  2
Total number of FSP-FPP relations in the tile: 0
Processing tile:  11
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1783762.9445651, 1785762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  11
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1785762.9445651, 1787762.9445651)
Number of segments interse

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 240
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-58993.2057939356, -58753.2057939356, 1787762.9445651, 1788002.9445651)
Processing tile:  13
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  13
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  13
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  6
Total number of FSP-FPP relations in the tile: 0
Processing tile:  13
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  13
Total number of FSP-FPP relations in the tile: 40105
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 156
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-61663.2057939356, -54533.2057939356, 1795812.9445651, 1800132.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-58613.2057939356, -57353.2057939356, 1796302.9445651, 1797762.9445651)
Processing tile:  14
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments inter

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Total number of FSP-FPP relations in the tile: 0
Processing tile:  15
Tile extent (minX, maxX, minY, maxY) : (-59353.2057939356, -57353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interseting with the tile:  2
Total number of FSP-FPP relations in the tile: 0
Processing tile:  15
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1783762.9445651, 1785762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  15
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1785762.9445651, 1787762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 2076
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-5

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 14150
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-57243.2057939356, -55713.2057939356, 1787762.9445651, 1789762.9445651)
Processing tile:  17
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 19204
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-56843.2057939356, -55353.2057939356, 1789762.9445651, 1791762.9445651)
Processing tile:  18
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  4


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 118
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-55483.2057939356, -55353.2057939356, 1791762.9445651, 1791972.9445651)
Processing tile:  19
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  9


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  19
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  15
Total number of FSP-FPP relations in the tile: 161966
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 506
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-60873.2057939356, -52053.2057939356, 1795412.9445651, 1799992.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-57353.2057939356, -55353.2057939356, 1795892.9445651, 1797762.9445651)
Processing tile:  20
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  8
Total number of FSP-FPP relations in the tile: 9633
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 8
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-60093.2057939356, -55793

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Total number of FSP-FPP relations in the tile: 0
Processing tile:  21
Tile extent (minX, maxX, minY, maxY) : (-57353.2057939356, -55353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  21
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1783762.9445651, 1785762.9445651)
Number of segments interseting with the tile:  1
Total number of FSP-FPP relations in the tile: 0
Processing tile:  21
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1785762.9445651, 1787762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 312
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-54

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53583.2057939356, -53353.2057939356, 1787762.9445651, 1789762.9445651)
Processing tile:  23
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 28316


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-55353.2057939356, -53353.2057939356, 1789762.9445651, 1791762.9445651)
Processing tile:  24
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  4


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 23572
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-55353.2057939356, -53353.2057939356, 1791762.9445651, 1792832.9445651)
Processing tile:  25
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  9


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 27766
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 249
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-55813.2057939356, -48293.2057939356, 1793442.9445651, 1796822.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-55023.2057939356, -53353.2057939356, 1794912.9445651, 1795762.9445651)
Processing tile:  26
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  12


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 67296
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 261
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-57333.2057939356, -48293.2057939356, 1793442.9445651, 1797032.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-55353.2057939356, -53353.2057939356, 1795762.9445651, 1796962.9445651)
Processing tile:  27
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  5
Total number of FSP-FPP relations in the tile: 0
Processing tile:  27
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  2
Total number of FSP-FPP relations in the tile: 0
Processing tile:  27
Tile extent (minX, maxX, minY, maxY) : (-55353.2057939356, -53353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments inters

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 33
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 1
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-48133.2057939356, -48123.2057939356, 1792952.9445651, 1792962.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53353.2057939356, -53293.2057939356, 1787642.9445651, 1787762.9445651)
Processing tile:  28
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 29540
Saving FSP-FPP relations in a file...


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep cur

Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53353.2057939356, -51353.2057939356, 1787762.9445651, 1789762.9445651)
Processing tile:  29
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 60174
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 4
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53353.2057939356, -51353.2057939356, 1789762.9445651, 1791762.9445651)
Processing tile:  30
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments i

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 19031
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 4
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53353.2057939356, -51353.2057939356, 1791762.9445651, 1792862.9445651)
Processing tile:  31
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  9


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 45929
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 186
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-55813.2057939356, -48293.2057939356, 1793442.9445651, 1796822.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53353.2057939356, -51353.2057939356, 1793952.9445651, 1795762.9445651)
Processing tile:  32
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  9


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 48770
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 318
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-54523.2057939356, -48293.2057939356, 1793442.9445651, 1796102.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-53353.2057939356, -51353.2057939356, 1795762.9445651, 1797042.9445651)
Processing tile:  33
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  2
Total number of FSP-FPP relations in the tile: 0
Processing tile:  33
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  33
Tile extent (minX, maxX, minY, maxY) : (-53353.2057939356, -51353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments inters

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  33
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 58597
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-51353.2057939356, -49353.2057939356, 1788182.9445651, 1789762.9445651)
Processing tile:  34
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 114519
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 4
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-51353.2057939356, -49353.2057939356, 1789762.9445651, 1791762.9445651)
Processing tile:  35
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  4


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 110243
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 8
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-51363.2057939356, -48123.2057939356, 1792952.9445651, 1794352.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-51353.2057939356, -49353.2057939356, 1791762.9445651, 1793762.9445651)
Processing tile:  36
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  5


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 73294
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 396
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-52093.2057939356, -48293.2057939356, 1793442.9445651, 1795882.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-51353.2057939356, -49353.2057939356, 1793762.9445651, 1795762.9445651)
Processing tile:  37
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  5


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 1585
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 7
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-52023.2057939356, -48293.2057939356, 1793442.9445651, 1795882.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-51353.2057939356, -49353.2057939356, 1795762.9445651, 1796372.9445651)
Processing tile:  38
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  38
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  38
Tile extent (minX, maxX, minY, maxY) : (-51353.2057939356, -49353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interseti

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  38
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 2374
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-49353.2057939356, -48753.2057939356, 1789282.9445651, 1789762.9445651)
Processing tile:  39
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 69265
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -48123.2057939356, 1792952.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-49353.2057939356, -47733.2057939356, 1789762.9445651, 1791762.9445651)
Processing tile:  40
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 102454
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 104
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1794332.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-49353.2057939356, -47393.2057939356, 1791762.9445651, 1793762.9445651)
Processing tile:  41
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 114054
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 99
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50783.2057939356, -47973.2057939356, 1792812.9445651, 1794352.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-49353.2057939356, -47353.2057939356, 1793762.9445651, 1795762.9445651)
Processing tile:  42
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments interseting with the tile:  3


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 30470
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50783.2057939356, -48293.2057939356, 1793442.9445651, 1794262.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-49353.2057939356, -47773.2057939356, 1795762.9445651, 1796622.9445651)
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-49353.2057939356, -47353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interset

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  43
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 71468
Saving FSP-FPP relation

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 10142
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 4
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1794262.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-46623.2057939356, -45353.2057939356, 1795762.9445651, 1796822.9445651)
Processing tile:  45
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  45
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  45
Tile extent (minX, maxX, minY, maxY) : (-47353.2057939356, -45353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interset

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  45
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  45
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  45
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 6932
Saving FSP-FPP relations in a file...


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-44373.2057939356, -43353.2057939356, 1793132.9445651, 1793762.9445651)
Processing tile:  46
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 111486
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-45353.2057939356, -43353.2057939356, 1793762.9445651, 1795762.9445651)
Processing tile:  47
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments 

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 54722
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-45353.2057939356, -43353.2057939356, 1795762.9445651, 1797402.9445651)
Processing tile:  48
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  48
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  48
Tile extent (minX, maxX, minY, maxY) : (-45353.2057939356, -43353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interset

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  48
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  48
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  48
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 34241
Saving FSP-FPP relations in a file...


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-43353.2057939356, -41353.2057939356, 1792072.9445651, 1793762.9445651)
Processing tile:  49
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 119864
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-43353.2057939356, -41353.2057939356, 1793762.9445651, 1795762.9445651)
Processing tile:  50
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments 

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 90628
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-43353.2057939356, -41353.2057939356, 1795762.9445651, 1797652.9445651)
Processing tile:  51
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  51
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  51
Tile extent (minX, maxX, minY, maxY) : (-43353.2057939356, -41353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interset

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 0
Processing tile:  51
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1787762.9445651, 1789762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  51
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1789762.9445651, 1791762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 0
Processing tile:  51
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1791762.9445651, 1793762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 46639
Saving FSP-FPP relations in a file...


E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-41353.2057939356, -39543.2057939356, 1791932.9445651, 1793762.9445651)
Processing tile:  52
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1793762.9445651, 1795762.9445651)
Number of segments interseting with the tile:  3
Total number of FSP-FPP relations in the tile: 78767
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-41353.2057939356, -39783.2057939356, 1793762.9445651, 1795762.9445651)
Processing tile:  53
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1795762.9445651, 1797762.9445651)
Number of segments i

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


Total number of FSP-FPP relations in the tile: 35470
Saving FSP-FPP relations in a file...
Number of unique FSPs in the tile: 3
Tile FSP extent (fspMinX,fspMaxX,fspMinY,fspMaxY):  (-50663.2057939356, -47973.2057939356, 1792812.9445651, 1793932.9445651)
Tile FPP extent (fppMinX,fppMaxX,fppMinY,fppMaxY):  (-41353.2057939356, -40233.2057939356, 1795762.9445651, 1797262.9445651)
Processing tile:  54
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1797762.9445651, 1799762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  54
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1799762.9445651, 1801762.9445651)
Number of segments interseting with the tile:  0
Total number of FSP-FPP relations in the tile: 0
Processing tile:  54
Tile extent (minX, maxX, minY, maxY) : (-41353.2057939356, -39353.2057939356, 1801762.9445651, 1803762.9445651)
Number of segments interset

E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))
E:\CUAHSI_SI/training/source\fldpln_library.py:198: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  fspDf = tdf.groupby(['FspId'], as_index=False).agg(MinDtf = ('Dtf', min),MaxDtf = ('Dtf', max))


## Calculate FSP and Segment Downstream Distance

Here we calculate downstream distance for the both FSPs and segments for mapping. It involves the following main tasks:
* Clean up segments 
    * It removes segments from the segment table if they are not in the FSP table. 
    * If a removed segment is the downstream segment of another segment in the segment table, the upstream segment ID is set to 0 (i.e., watershed outlet). 
    * Those removed segments are usually close to or in waterbodies. By removing those segments, a library may have several separate watersheds/outlets! For example, neosho has 3 separate watersheds (segment 13, 104, 186 as the outlet segments). 
* Calculate FSP and segment downstream distance (i.e., distance from outlet) using in interpolating FSP depth of water from gauges

At the end, this step updates the FSP and segment info CSV files with additional columns. 

In [20]:
from fldpln_library import *
from fldpln import *

In [21]:
# tiled library folder
tiledLibFolder = r'E:\CUAHSI_SI\training\examples\wildcat_10m_3dep\tiled_snz_library' 

# libraries to be tiled. 
# Note that the tiled libraries will have the same name as the segment-based libraries except they are located under the tiledLibFolder!
libNames = ['lib_py'] # libs with different segments and fldmx in Wildcat Creek

# Create FSP and segment info files and library meta file (i.e., cell size and spatial reference)
for libName in libNames:
    print(f'Calculate FSP and segment downstream distance: {libName} ...')
    CalculateFspSegmentDownstreamDistance(tiledLibFolder,libName)

Calculate FSP and segment downstream distance: lib_py ...


## Assign Stream Order to FSPs and Segments Manually

Stream orders are used while interpolating the depth of flow (DOF) at FSPs where low order streams are handled before high order streams. Currently this is performed in GIS manually by first create the segment shapefiles from the FSP and segment info CSV files. 

### Generate Segment Shapefile

Create a shapefile for manually assigning stream orders to segments in a tiled library. Note that the shapefile gets its CRS from the library metadata file and has all the attributes in the FSP and segment info CSV file.

In [22]:
# tiled library folder to generate segment shapefile
tiledLibFolder= r'E:\CUAHSI_SI\training\examples\wildcat_10m_3dep\tiled_snz_library\lib_py'

# FSP and segment info CSV files
segInfoFile = os.path.join(tiledLibFolder,segInfoFileName)
fspInfoFile = os.path.join(tiledLibFolder,fspInfoFileName)

# Read lib CRS from metadata file
metaDataFile = os.path.join(tiledLibFolder,metaDataFileName)
with open(metaDataFile,'r') as jf:
    md = json.load(jf)
srText = md['SpatialReference']
libCrs = CRS.from_wkt(srText)

# segment shapefile folder
# shpFolder = r'D:\xingong\Wildcat\stream_order'
shpFolder = tiledLibFolder
shpName = "segments1.shp"
outShpFile =  os.path.join(shpFolder,shpName)

# generate segment shapefile 
print(f'Generate segment shapefile for libraries ...')
# GenerateSegmentShapefiles(tiledLibFolder,libNames,shpFolder,shpName)
GenerateSegmentShapefilesFromFspSegmentInfoFiles(segInfoFile, fspInfoFile, libCrs, outShpFile)

Generate segment shapefile for libraries ...


### Assign Stream Orders to Segment Shapefile in GIS

Here we assign stream order to shapefile manually in GIS. We need to create a field, say 'StrOrd', in the shapefile and then select segments and assign stream order value to the segmentd.

The general rules for assigning stream order are:
* A stream (or reach) consists of several connected segments, which may meet with other streams at confluences
* Main streams have low orders than tributary streams
* Order must be unique to each stream in a library. No two streams can have the same order even if the streams are in different sub-watersheds in the library

The above requirements are necessary to make sure the interpolation on DOF can be carried out correctly.


### Get Stream Order to FSPs and Segments from Segment Shapefile

This step gets stream orders from the segment shapefile and add them to the FSP and segment info files. 

It also creates a new text file, stream_order_info.csv, which stores the connectivity among stream orders with columns: [‘StrOrd’, ‘DsStrOrd’, ‘JunctionFspX’, ‘JunctionFspY’]. This information is used in DOF interpolation.

In [23]:
# tiled library folder to add stream order
tiledLib= r'E:\CUAHSI_SI\training\examples\wildcat_10m_3dep\tiled_snz_library\lib_py'

# FSP and segment info CSV files
segInfoFile = os.path.join(tiledLib,segInfoFileName)
fspInfoFile = os.path.join(tiledLib,fspInfoFileName)

# segment shapefile which has the stream order
shpFolder = r'E:\CUAHSI_SI\training\examples\wildcat_10m_3dep\tiled_snz_library\lib_py'
shpName = "segments1.shp"
shpOrdColName = 'str_ord'
shpFile =  os.path.join(shpFolder,shpName)

print(f'Get stream order and generate stream order network info for : {tiledLib} ...')
GetStreamOrdersForFspsSegments(tiledLib,shpFile,shpOrdColName)

Get stream order and generate stream order network info for : E:\CUAHSI_SI\training\examples\wildcat_10m_3dep\tiled_snz_library\lib_py ...


E:\CUAHSI_SI/training/source\fldpln_library.py:622: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  strOrdDf = pd.concat([strOrdDf, temp],ignore_index=True)


(      FspId          FspX          FspY  SegId  FilledElev        DsDist  \
 0         1 -61658.205794  1.800128e+06      1  342.556946  31350.024510   
 1         2 -61648.205794  1.800118e+06      1  342.512115  31335.882374   
 2         3 -61648.205794  1.800108e+06      1  342.512115  31325.882374   
 3         4 -61638.205794  1.800098e+06      1  342.512115  31311.740239   
 4         5 -61628.205794  1.800088e+06      1  342.512115  31297.598103   
 ...     ...           ...           ...    ...         ...           ...   
 2653   2654 -48018.205794  1.792858e+06     16  302.513397     56.568542   
 2654   2655 -48008.205794  1.792848e+06     16  302.507172     42.426407   
 2655   2656 -47998.205794  1.792838e+06     16  302.501190     28.284271   
 2656   2657 -47988.205794  1.792828e+06     16  302.495056     14.142136   
 2657   2658 -47978.205794  1.792818e+06     16  302.488861      0.000000   
 
       StrOrd  
 0        1.0  
 1        1.0  
 2        1.0  
 3        

## Assign Stream Order to FSPs and Segments Automatically

* This is currently been implemented by Junho @ UA
* Stream order should be decided by the length and flow accumulation
* The level path approach used in NWC may be used here to automate stream order assignment!

In [ ]:
# Code to automatically assign stream order to segments based on stream length or flow accumulation

### Generate Segment Shapefile

Here we generate segment shapefile to check automatic stream order assignment.

## Distribute Tiled Library

Now we can make tiled libraries available for flood inundation mapping. This involves:
* Zip all the files for each tiled library as a single zip file. The code below can be used to automate this step
* Update scripts/package and notebooks
* Update example libraries and gauge files

All the files can be distributed through KU FTP site. A better option is to distribute libraries at KU server but update the package and examples on GitHub. 

In [ ]:
# library folder
libFolder = r'D:\xingong\fldpln_3dep_vg\tiled10km_snz_library'
# ftp folder
ftpFolder =  r'D:\inetpub\wwwroot\download\fldpln\libraries\verdigris_10m'

# libraries to zip
libNames = ['verdigris']

# zip libraries
for libName in libNames:
    print(f'Zipping library: {libName} ...')
    shutil.make_archive(os.path.join(ftpFolder,libName), 'zip', os.path.join(libFolder,libName))